### PR TITLE
Implemented mouse wheel callback support for Linux

### PR DESCRIPTION
**Pull request by theg4sh**
The changes made possible to make custom handler on mouse wheel
events in QT and GTK for Linux opencv build.

_Reopened PR https://github.com/opencv/opencv/pull/6725 due merge checking problem_

### CODE REVIEW

**Comment by vpisarev** [comment]
thanks for the contribution! :+1:

File: modules/highgui/src/window_QT.cpp [hunk 1]
```diff
@@ -2629,8 +2629,18 @@ void DefaultViewPort::resizeEvent(QResizeEvent* evnt)
 
 void DefaultViewPort::wheelEvent(QWheelEvent* evnt)
 {
-    scaleView(evnt->delta() / 240.0, evnt->pos());
+    int delta = evnt->delta();
+    int cv_event = -1;
+    int flags = ((delta & 0xffff)<<16) | ((evnt->orientation() == Qt::Vertical) ? CV_EVENT_MOUSEWHEEL : CV_EVENT_MOUSEHWHEEL);
+    QPoint pt = evnt->pos();
+
+    icvmouseHandler(evnt, mouse_wheel, cv_event, flags);
```
Comment (ilya-lavrenov) [inline modules/highgui/src/window_QT.cpp]: `mouse_wheel` is undeclared identifier. @theg4sh, have you ever tested this code?

Comment (theg4sh) [inline modules/highgui/src/window_QT.cpp]: Well, it was here but removed here: https://github.com/opencv/opencv/pull/6787/commits/fd78e2cc6cdcbf7844dc38fb91d05922739f8bd9
and I forgot to update the code. I didn't tested with QT yet. Thanks for response, I'm working on it.

Comment (theg4sh) [inline modules/highgui/src/window_QT.cpp]: PR on the way. Ilya, please, look for https://github.com/opencv/opencv/pull/6966

**Pull request merged**